aws/resource_aws_cloudfront_realtime_log_config.go

```diff
@@ -0,0 +1,245 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudfront"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/cloudfront/finder"
+)
+
+func resourceAwsCloudFrontRealtimeLogConfig() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCloudFrontRealtimeLogConfigCreate,
+		Read:   resourceAwsCloudFrontRealtimeLogConfigRead,
+		Update: resourceAwsCloudFrontRealtimeLogConfigUpdate,
+		Delete: resourceAwsCloudFrontRealtimeLogConfigDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"endpoint": {
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"stream_type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"Kinesis"}, false),
+						},
+
+						"kinesis_stream_config": {
+							Type:     schema.TypeList,
+							Required: true,
+							MinItems: 1,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"role_arn": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validateArn,
+									},
+
+									"stream_arn": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validateArn,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"fields": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"sampling_rate": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.IntBetween(1, 100),
+			},
+		},
+	}
+}
+
+func resourceAwsCloudFrontRealtimeLogConfigCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudfrontconn
+
+	input := &cloudfront.CreateRealtimeLogConfigInput{
+		EndPoints:    expandCloudFrontEndPoints(d.Get("endpoint").([]interface{})),
+		Fields:       expandStringSet(d.Get("fields").(*schema.Set)),
+		Name:         aws.String(d.Get("name").(string)),
+		SamplingRate: aws.Int64(int64(d.Get("sampling_rate").(int))),
+	}
+
+	log.Printf("[DEBUG] Creating CloudFront Real-time Log Config: %s", input)
+	output, err := conn.CreateRealtimeLogConfig(input)
+
+	if err != nil {
+		return fmt.Errorf("error creating CloudFront Real-time Log Config: %w", err)
```

Comment:
nit: for more context, return the name. might be easier to store the name initially in var up top

```suggestion
		return fmt.Errorf("error creating CloudFront Real-time Log Config (%s): %w", d.Get("name").(string), err)
```

Reply:
In new commit (added `name` variable).